com.unity.ml-agents.extensions/Editor/GridSensorComponentEditor.cs

```diff
@@ -0,0 +1,115 @@
+using UnityEditor;
+using UnityEngine;
+using Unity.MLAgents.Editor;
+using Unity.MLAgents.Extensions.Sensors;
+
+namespace Unity.MLAgents.Extensions.Editor
+{
+    [CustomEditor(typeof(GridSensorComponent))]
+    [CanEditMultipleObjects]
+    internal class GridSensorComponentEditor : UnityEditor.Editor
+    {
+        public override void OnInspectorGUI()
+        {
+            var so = serializedObject;
+            so.Update();
+
+            // Drawing the GridSensorComponent
+            EditorGUI.BeginChangeCheck();
+
+            EditorGUI.BeginDisabledGroup(!EditorUtilities.CanUpdateModelProperties());
+            {
+                // These fields affect the sensor order or observation size,
+                // So can't be changed at runtime.
+                EditorGUILayout.PropertyField(so.FindProperty("m_SensorName"), true);
+
+                EditorGUILayout.LabelField("Grid Settings", EditorStyles.boldLabel);
+                EditorGUILayout.PropertyField(so.FindProperty("m_CellScaleX"), true);
+                EditorGUILayout.PropertyField(so.FindProperty("m_CellScaleZ"), true);
```

Comment:
X, Y, Z instead of X, Z, Y? I understand it's probably that way because (X, Z) are the plane that the grid is in, but it's counter-intuitive.

Reply:
Also, just for screen real estate, can we see if it's easy put the cell scale and grid num size fields on the same line? Maybe user a Vector3 for the scale?

Reply:
and Vector2Int for the GridNumSide

Reply:
I was going to change these name to something like length, width, height later.
But sure I can make them Vector3 and Vector2Int

Reply:
I ended up making GridNumSide Vector3/Vector3Int and is only configurable on x and z, in case we want 3D GridSensor one day.